### PR TITLE
fix: updated solution for Convert JSON Data to HTML

### LINKED ISSUE #41133: Convert JSON Data to HTML regex typo
Challenge link: https://www.freecodecamp.org/learn/data-visualization/json-apis-and-ajax/convert-json-data-to-html

Code that fails:
```
        // Add your code below this line
        json.forEach(function(val) {
          // Adding each object keys
          var keys = Object.keys(val);
          // Generating new html
          html+= "<div class = 'cat'>";
          // Adding the custom html to each key
          keys.map(function(key) {
            html+= "<strong>" + key + "</strong>: " + val[key] + "<br>";
          });
          html+= "</div><br>";
        });
        // Add your code above this line
```
Code that passes:
```
        // Add your code below this line
        json.forEach(function(val) {
          // Adding each object keys
          var keys = Object.keys(val);
          // Generating new html
          html+= "<div class = 'cat'>";
          // Adding the custom html to each key
          keys.map(function(key) {
            html += "<strong>" + key + "</strong>: " + val[key] + "<br>";
          });
          html+= "</div><br>";
        });
        // Add your code above this line
```

Issue: `assert(code.match(/html\s+?(\+=|=\shtml\s\+)/g));`

Correction: `assert(__helpers.removeWhiteSpace(code).match(html(\+=|=html\+))`

Challenge file: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.md

### PR DESCRIPTION
**Pull request by fotscode**
Checklist:

- [ x ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [ x ] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [ x ] My pull request targets the `main` branch of freeCodeCamp.
- [ x ] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #41133 

Resolves issue [#41133 ](https://github.com/freeCodeCamp/freeCodeCamp/issues/41133)

### CODE REVIEW

File: curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Here is some example JSON
 Your code should store the data in the `html` variable
 
 ```js
-assert(code.match(/html\s+?(\+=|=\shtml\s\+)/g));
+assert(__helpers.removeWhiteSpace(code).match(html(\+=|=html\+))
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.md:58]:
```suggestion
assert(__helpers.removeWhiteSpace(code).match(/html(\+=|=html\+/g))
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
This was partially my fault, but you can use the `commit suggestion` button to commit this change

**Comment by ShaunSHamilton** [comment]
Also, for future reference, in GitHub, words like `Closes ` are keywords which used to help link things like issues and PRs. So, when you create a PR that resolves an existing issue, you can automatically close the issue (once the PR is merged), by including `Closes #XXXXX` in the PR body. (where `XXXXX` is the PR number.

I have edited your PR body to include this.

File: curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Here is some example JSON
 Your code should store the data in the `html` variable
 
 ```js
-assert(code.match(/html\s+?(\+=|=\shtml\s\+)/g));
+assert(__helpers.removeWhiteSpace(code).match(/html(\+=|=html\+/g))
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.md:58]:
```suggestion
assert(__helpers.removeWhiteSpace(code).match(/html(\+=|=html\+)/g))
```
This should fix things.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]

**Review by RandellDawson — APPROVED** [review]
LGTM

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by ojeytonwilliams** [comment]
@ShaunSHamilton if you're happy, I think this is good to merge.

**Comment by ShaunSHamilton** [comment]
Congrats, on your first pull request (PR)! 🎉

Thank you, for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Review by ShaunSHamilton — APPROVED** [review]

**Pull request merged**